clusterloader/framework/util.go

```diff
@@ -17,66 +17,130 @@ limitations under the License.
+	existingNamespaces, _ := f.ClientSet.Core().Namespaces().List(metav1.ListOptions{})
+	for _, value := range existingNamespaces.Items {
+		if value.GenerateName == fmt.Sprintf("e2e-tests-%v-", baseName) {
+			return value.Name
 		}
-		if tuning != nil {
-			// If a rate limit has been defined we wait for N ms between creation
-			if tuning.Pods.RateLimit.Delay != 0 {
-				framework.Logf("Sleeping %d ms between podcreation.", tuning.Pods.RateLimit.Delay)
-				time.Sleep(tuning.Pods.RateLimit.Delay * time.Millisecond)
-			}
-			// If a stepping tuningset has been defined in the config, we wait for the step of pods to be created, and pause
-			if tuning.Pods.Stepping.StepSize != 0 && (i+1)%tuning.Pods.Stepping.StepSize == 0 {
-				verifyRunning := f.NewClusterVerification(
-					&v1.Namespace{
-						ObjectMeta: metav1.ObjectMeta{
-							Name: ns,
-						},
-						Status: v1.NamespaceStatus{},
-					},
-					framework.PodStateVerification{
-						Selectors:   labels,
-						ValidPhases: []v1.PodPhase{v1.PodRunning},
-					},
-				)
+	}
+	return ""
+}
 
-				pods, err := verifyRunning.WaitFor(i+1, tuning.Pods.Stepping.Timeout*time.Second)
-				if err != nil {
-					framework.Failf("Error in wait... %v", err)
-				} else if len(pods) < i+1 {
-					framework.Failf("Only got %v out of %v", len(pods), i+1)
-				}
+// ParseConfig unmarshalls the json file defined in the CL config into a struct
+func (cl *ClusterLoaderObject) ParseConfig() (*v1.Pod, error) {
+	pod := &v1.Pod{}
+	// If the file is defined used that as the config
+	if cl.File != "" {
+		configFile, err := ioutil.ReadFile(MakePath(cl.File))
+		if err != nil {
+			return pod, err
+		}
 
-				framework.Logf("We have created %d pods and are now sleeping for %d seconds", i+1, tuning.Pods.Stepping.Pause)
-				time.Sleep(tuning.Pods.Stepping.Pause * time.Second)
-			}
+		if err = json.Unmarshal(configFile, &pod); err != nil {
+			return pod, err
+		}
+	} else if cl.Image != "" && cl.Basename != "" {
+		// Otherwise if we have the image name use that instead
+		zero := int64(0)
+		pod.Spec = v1.PodSpec{TerminationGracePeriodSeconds: &zero, Containers: []v1.Container{
+			{
+				Name:  cl.Basename,
+				Image: cl.Image,
+			},
+		},
 		}
+	} else {
+		return pod, errors.New("Missing both config file and imagename")
+	}
+
+	return pod, nil
+}
+
+// MakePath returns fully qualfied file location as a string
+func MakePath(file string) string {
+	// Handle an empty filename.
+	if file == "" {
+		framework.Failf("No template file defined!")
+	}
+	return filepath.Join("content/", file)
+}
+
+// ConvertToLabelSet will convert the string label to a set, while also setting a default value
+func (cl *ClusterLoaderObject) ConvertToLabelSet() (labels.Set, error) {
+	if cl.Label == "" {
+		cl.Label = "purpose=test"
+	}
+	label, err := labels.ConvertSelectorToLabelsMap(cl.Label)
+	return label, err
+}
+
+// TimeSplit will take the user readable time string and convert it into corresponding time.Duration
```

Comment:
As I wrote above - remove this function and replace with ParseDuration:
https://golang.org/pkg/time/#ParseDuration